workspaces/ballerina/ballerina-extension/src/features/ai/service/libs/langlibs.ts

```diff
@@ -14,7 +14,377 @@
+\`\`\`
+
+**Converting between types with default values**, use \`cloneWithType()\`. Uses **dependent types**.
+\`\`\`ballerina
+json cfg = {port: 8080};
+type Config record {| int port; int timeout = 60; |};
+Config c = check cfg.cloneWithType();
+
+anydata[] arr = [1, 2, 3];
+int[] nums = check arr.cloneWithType(int[]);
+\`\`\`
+
+**Checking if a value matches a specific type**, use \`ensureType()\`. Uses **dependent types** - validates without copying, and the return type matches your \`typedesc\` parameter.
+\`\`\`ballerina
+json student = {name: "Jo", subjects: ["CS1212"]};
+json[] subs = check student.subjects.ensureType();
+
+anydata data = "hello";
+string s = check data.ensureType(string); 
+\`\`\`
+
+**Turning any value into a readable string**, use \`toString()\`. Great for logging and showing users.
+\`\`\`ballerina
+map<int> scores = {Alice: 90, Bob: 85};
+string output = scores.toString();
+\`\`\`
+
+**Converting data to JSON format**, use \`toJson()\`. Perfect for API responses and saving data.
+\`\`\`ballerina
+record {string name; int age;} person = {name: "Alice", age: 30};
+json result = person.toJson(); 
+\`\`\`
+
+**Turning data into a JSON string**, use \`toJsonString()\`. Use this when you need to store or send JSON as text.
+\`\`\`ballerina
+map<int> scores = {Alice: 90, Bob: 85};
+string jsonText = scores.toJsonString();
+\`\`\`
+
+**Parsing JSON text into a json value**, use \`fromJsonString()\`. Great for reading JSON from files or APIs.
+\`\`\`ballerina
+string jsonText = "{\"id\":12,\"name\":\"Alice\"}";
+json data = check jsonText.fromJsonString();
+\`\`\`
+
+**Converting JSON into a specific type**, use \`fromJsonWithType()\`. Uses **dependent types** - return type matches your \`typedesc\` parameter.
+\`\`\`ballerina
+json jsonArr = [1, 2, 3];
+int[] nums = check jsonArr.fromJsonWithType();
+
+json cfg = {port: 8080, timeout: 60};
+type Config record {| int port; int timeout; |};
+Config config = check cfg.fromJsonWithType(Config);
+\`\`\`
+
+**Parsing JSON text directly into a typed value**, use \`fromJsonStringWithType()\`. Uses **dependent types** - return type matches your \`typedesc\` parameter.
+\`\`\`ballerina
+string jsonText = "[1, 2, 3]";
+int[] nums = check jsonText.fromJsonStringWithType();
+
+string configText = "{\"port\":8080,\"timeout\":60}";
+type Config record {| int port; int timeout; |};
+Config config = check configText.fromJsonStringWithType(Config); 
+\`\`\`
+
+## What You Can Do with Arrays
+
+- **Counting elements**, use \`length()\`.
+- **Transforming each element**, use \`map()\` with your transformation function. Returns a fresh array.
+- **Picking elements that match criteria**, use \`filter()\` with a predicate function. Returns a fresh array.
+- **Running an operation on each element**, use \`forEach()\` with an iterator function for side effects. Returns ().
+- **Reducing all elements into one value**, use \`reduce()\` with a combiner function and starting value.
```

Comment:
These are lambda functions, we can introduce this when BI supprots them, we dont need it for now